test/support/assert_api_response.rb

```diff
@@ -30,6 +30,53 @@ def assert_api_request(method, path, params = {})
 
     assert_mock http
   end
+
+  # Asserts that the given resource can use auto-pagination.
+  #
+  # path     - API path (without `/api/v1` prefix, eg: "/messages")
+  # resource - ReamazeAPI resource name (eg: `:messages`)
+  #
+  # Example
+  #
+  #   assert_pagination "/conversations", :conversations
+  def assert_pagination(path, resource)
```

Comment:
i see the expectation setting up for a page size of 2, but i don't see how that is setup to be expected otherwise - it feels like its just testing that it's correctly stubbed? I'm not clear on how this tests the code paginates correctly - but i admit i don't fully get whats going on either.